src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -479,6 +479,7 @@ public class Options
 
             /// <summary>
             /// Whether the data file has a header with feature names.
+            /// Note: If a TextLoader is created with hasHeader=true but without a dataSample, then the TextLoader will not contain slot names, because the output schema is made when the TextLoader is made, and not when Load is called.
             /// </summary>
```

Comment:
Can you please explain the reference to slots? Does it mean column names?
Is it possible to throw an exception instead under these conditions?

Reply:
Of course. Yes, the term "slots" here refers to columns that are chosen for data manipulation. In my latest commit I have added the case where a situation where hasHeader = true and null dataSample results in a Not Supported exception.